sdk/js/index.ts

```diff
@@ -272,32 +283,34 @@ export default class StreamManager {
   }
 
   onStreamCreated(
-    handleStream: (stream: Stream) => null,
+    handleStream: (stream: Stream) => void,
     creator?: Address
   ): void {
-    const onLogs = (logs: Log[]) => {
-      logs
-        .map(
-          // Log is StreamCreated
-          (log: Log) =>
-            new Stream(
-              this,
-              log.topics[2] as Address, // creator
-              Number(log.topics[3]), // streamId
-              log.topics[4] as Address, // token
-              BigInt(log.topics[5] as string), //amount per second
-              this.publicClient,
-              this.walletClient
-            )
-        )
-        .forEach(handleStream);
-    };
-    this.publicClient.watchContractEvent({
-      address: this.address,
-      abi: StreamManagerContractType.abi as Abi,
-      eventName: "StreamCreated",
-      args: creator ? { creator } : {},
-      onLogs,
-    });
+    try {
+      this.publicClient.watchContractEvent({
+        address: this.address,
+        abi: StreamManagerContractType.abi as Abi,
+        eventName: "StreamCreated",
+        args: creator ? { creator } : {},
+        onLogs: (logs: Log[]) => {
+          (logs as StreamCreated[])
```

Comment:
Wonder if this would work
```suggestion
        onLogs: (logs: StreamCreated[]) => {
          logs
```

Reply:
it does!